core/src/test/java/org/apache/iceberg/ScanTestBase.java

```diff
@@ -19,114 +19,104 @@
 package org.apache.iceberg;
 
 import static org.apache.iceberg.types.Types.NestedField.required;
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assumptions.assumeThat;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
 import java.util.Collections;
 import java.util.concurrent.Executors;
 import java.util.concurrent.atomic.AtomicInteger;
 import org.apache.iceberg.expressions.Expression;
 import org.apache.iceberg.expressions.Expressions;
 import org.apache.iceberg.io.CloseableIterable;
-import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.types.Types;
-import org.assertj.core.api.Assertions;
-import org.junit.Assert;
-import org.junit.Assume;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-@RunWith(Parameterized.class)
+@ExtendWith(ParameterizedTestExtension.class)
 public abstract class ScanTestBase<
         ScanT extends Scan<ScanT, T, G>, T extends ScanTask, G extends ScanTaskGroup<T>>
-    extends TableTestBase {
-
-  @Parameterized.Parameters(name = "formatVersion = {0}")
-  public static Object[] parameters() {
-    return new Object[] {1, 2};
-  }
-
-  public ScanTestBase(int formatVersion) {
-    super(formatVersion);
-  }
+    extends ParameterizedTableTestBase {
```

Comment:
you should be able to extend `TestBase` now

Reply:
extended with `TestBase`, still getting similar errors. 

`DeleteFileIndexTestBase`, `SparkDistributedDataScanTestBase`, `DataTableScanTestBase`, `ScanPlanningAndReportingTestBase` are all `core` files that now extend from `TestBase`, and are also used in older Spark versions. We probably need both JUnit4 and JUnit5 versions of these files to co-exist. 

I can make that a separate PR if you think it's the right approach.